hoodie-client/src/main/java/com/uber/hoodie/index/hbase/HBaseIndex.java

```diff
@@ -303,14 +328,118 @@ private void doPutsAndDeletes(HTable hTable, List<Put> puts, List<Delete> delete
     hTable.flushCommits();
     puts.clear();
     deletes.clear();
+    sleepForTime(sleepTimeMsForPutBatch);
+  }
+
+  private static void sleepForTime(int sleepTimeMs) {
+    try {
+      Thread.sleep(sleepTimeMs);
+    } catch (InterruptedException e) {
+      logger.error(e);
+      throw new RuntimeException(e);
+    }
   }
 
   @Override
   public JavaRDD<WriteStatus> updateLocation(JavaRDD<WriteStatus> writeStatusRDD, JavaSparkContext jsc,
       HoodieTable<T> hoodieTable) {
+    setPutBatchSize(writeStatusRDD, jsc);
     return writeStatusRDD.mapPartitionsWithIndex(updateLocationFunction(), true);
   }
 
+  private void setPutBatchSize(JavaRDD<WriteStatus> writeStatusRDD,
+      final JavaSparkContext jsc) {
+    if (config.getHbaseIndexPutBatchSizeAutoCompute()) {
+      SparkConf conf = jsc.getConf();
+      int maxExecutors = conf.getInt("spark.executor.instances", 1);
```

Comment:
These config names can change across spark versions and eventually break here. Can you find a way to avoid using these ?

Reply:
@n3nash  I can have hoodie configs to accept spark config names. Do you recommend me to move those config to HoodieWriteConfig or HoodieIndexConfig?

Reply:
Ideally we want to avoid using them if possible, can you check in the spark-context object gives back the information that you need via methods ? Else, let's move it to constants inside HbaseIndex.

Reply:
Done, couldn't find any public api to fetch num executors, so moving these to constants within HbaseIndex